### ISSUE TITLE
Support decoding into `OsString`s

### ISSUE THREAD

**Issue by BlacklightShining** [docopt/docopt.rs#152]
From what I can tell, docopt only supports parsing arguments into Rust strings. This means that if a program uses docopt, it won't be able to take any arguments that aren't valid UTF-8, even if they may be otherwise perfectly valid (e.g. a path to a file on a *nix system).

**Comment by BlacklightShining** [comment]
Related to #135, I think: the ability to take paths as `OsString`s would probably solve that, while avoiding ambiguity in interpreting whether a `Vec<u8>` is a `u8` option that can be passed multiple times, or a not-necessarily-valid-UTF-8 string.

**Comment by BurntSushi** [comment]
You are absolutely correct. I was not particularly principled with this when building docopt, so it is quite possible that this could require some refactoring.
